classes/local/v1p1/csv_client.php

```diff
@@ -136,6 +143,46 @@ public function set_data(
         ];
     }
 
+    /**
+     * Set the CSV data for synchronization with v1p2 additional fields.
+     *
+     * @param array $manifest The manifest data.
+     * @param array $users The users data.
+     * @param array $classes The classes data.
+     * @param array $courses The courses data.
+     * @param array $orgs The orgs data.
+     * @param array $enrollments The enrollments data.
+     * @param array $academicsessions The academic sessions data.
+     * @param array $roles The roles data.
+     * @param array $demographics The demographics data.
+     * @param array $userprofiles The user profiles data.
+     */
+    public function versioned_set_data(
+        array $manifest,
+        array $users,
+        array $classes,
+        array $courses,
+        array $orgs,
+        array $enrollments,
+        array $academicsessions,
+        array $roles,
+        array $demographics,
+        array $userprofiles
+    ): void {
+        $this->data = [
+            'manifest' => $manifest,
+            'users' => $users,
+            'classes' => $classes,
+            'courses' => $courses,
+            'orgs' => $orgs,
+            'enrollments' => $enrollments,
+            'academicSessions' => $academicsessions,
+            'roles' => $roles,
+            'demographics' => $demographics,
+            'userProfiles' => $userprofiles,
```

Comment:
This is the v1p1 client, why does it contain userProfiles? (I understand this to be a v1p2 only). Should this instead be nullable and an optional param?